Automattic/jetpack
# Improve accessibility on story block with keyboard shortcuts

This PR adds keyboard support for the story block and more generally improves accessibility on the block.
This is a follow up of https://github.com/Automattic/jetpack/pull/16852


#### Changes proposed in this Pull Request:

This is an important refactor of the Story Block as we're moving away from managing the player's state within the player component itself to using a global (redux) store. This allows the story player to be unmounted and remounted somewhere else on the page.

This should be fixing the z-index issues as we now have a true modal component (appended at the end of the DOM with a high z-index).

When displaying a story in a modal we're focusing on the story player itself as well as trapping the keyboard in order to prevent tabbing on other elements of the page. On exit we refocus on the inlined payer element.

While playing controls can be iterated using tab/shit+tab and img alt and title can be read by screen readers. This should offer a slightly better overall experience for screen reader users.

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
- Test the changes on a .org and wpcom site
- Make sure there is no regression regarding story loading and playback
- Try playing a story using your keyboard: tab to focus on player, space to play/pause, escape to leave fullscreen
- Try from an android device
- Try from an iphone
- Try using voiceover on your mac and iphone
- Try using talkback on android
